### PR TITLE
For IOS/UIKit Support UIGestureRecognizerDelegate for simultaneous gesture input as well as missing stock UIGestureRecognizers (#3413)

### PR DESCRIPTION
**Pull request by jpedrick**
Replaces https://github.com/rust-windowing/winit/pull/3421.

- [ x] Tested on all platforms changed
- [x ] Added an entry to `CHANGELOG.md` if knowledge of this change could be valuable to users
- [ ] Updated documentation to reflect any user-facing changes, including notes of platform-specific behavior
- [x ] Created or updated an example program if it would help users understand this functionality
- [ ] Updated [feature matrix](https://github.com/rust-windowing/winit/blob/master/FEATURES.md), if new features were added or implemented

### CODE REVIEW

File: src/platform_impl/ios/uikit/key.rs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    unsafe impl UIKey {
+        // https://developer.apple.com/documentation/uikit/uikey/3526132-keycode?language=objc
+        #[method(keyCode)]
+        pub fn key_code(&self) -> CFIndex; // -> enum UIKeyboardHIDUsage
+
+        // https://developer.apple.com/documentation/uikit/uikeymodifierflags?language=objc
+        #[method(modifierFlags)]
+        pub fn modifier_flags(&self) -> NSInteger; // -> enum UIKeyModifierFlags
+
+        #[method(characters)]
```
Comment (jpedrick) [inline src/platform_impl/ios/uikit/key.rs:26]: method -> method_id

```rust
#[method_id(characters)]
pub fn characters(&self) -> Id<NSString>;

#[method_id(charactersIgnoringModifiers)]
pub fn characters_ignoring_modifiers(&self) -> Id<NSString>;
```

File: src/platform_impl/ios/view.rs [hunk 2]
```diff
@@ -62,6 +77,12 @@ declare_class!(
             let _: () = unsafe { msg_send![super(self), drawRect: rect] };
         }
 
+        #[method(gestureRecognizer:shouldRecognizeSimultaneouslyWithGestureRecognizer:)]
+        fn should_recognize_simultaneously(&self, gesture_recognizer: &UIGestureRecognizer, other_gesture_recognizer: &UIGestureRecognizer) -> bool {
+            log::debug!("called class method should_recognize_simultaneously");
```
Comment (jpedrick) [inline src/platform_impl/ios/view.rs:82]: Remove debug log here

File: src/platform_impl/ios/view.rs [hunk 3]
```diff
@@ -270,6 +353,10 @@ impl WinitView {
             pinch_gesture_recognizer: RefCell::new(None),
             doubletap_gesture_recognizer: RefCell::new(None),
             rotation_gesture_recognizer: RefCell::new(None),
+            pan_gesture_recognizer: RefCell::new(None),
+            long_press_gesture_recognizer: RefCell::new(None),
+            // eventually allow passing this from WindowBuilder
```
Comment (jpedrick) [inline src/platform_impl/ios/view.rs:358]: Remove comment now that issue #3428 is created

**Comment by daxpedda** [comment]
@jpedrick would you mind giving this a proper title?

**Comment by jpedrick** [comment]
> @jpedrick would you mind giving this a proper title?

Done!

**Comment by VladasZ** [comment]
Great feature. Is there any progress on this? Maybe I can help?

**Comment by kchibisov** [comment]
we've added some gestures on ios recently, so maybe your needs are already covered.

cc @madsmtm

**Comment by VladasZ** [comment]
You talking about this: https://github.com/rust-windowing/winit/pull/3130? I need `PanGesture` and I see it in this PR.

**Comment by jpedrick** [comment]
> You talking about this: #3130? I need `PanGesture` and I see it in this PR.

I’ve been meaning to resubmit the PR for this with code I’ve been using in a private project for a couple of months. I’ll get to it tomorrow and hopefully we can get this moved forward.

**Pull request closed without merging**